plugins/bitbucket-cloud-common/src/BitbucketCloudConfig.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/** @public */
+export type BitbucketCloudConfig = {
+  /**
+   * Constant. https://api.bitbucket.org/2.0
+   */
+  baseUrl?: string;
+
+  auth?: {
+    /**
+     * The username to use for requests to Bitbucket Cloud (bitbucket.org).
+     */
+    username: string;
+
+    /**
+     * Authentication with Bitbucket Cloud (bitbucket.org) is done using app passwords.
+     *
+     * See https://support.atlassian.com/bitbucket-cloud/docs/app-passwords/
+     */
+    appPassword: string;
+  };
+};
```

Comment:
is this type necessary, in addition to https://github.com/backstage/backstage/blob/431c66316fc32596d80f43df171c85db2dba400c/packages/integration/src/bitbucketCloud/config.ts#L27 ?

Reply:
Yeah looking at the code in the client above, the integrations already have the default API url mechanism and stuff, should just pass in an instance of BitbucketCloudIntegrationConfig to that instead of this duplicate interface, I think

Reply:
this would couple the bitbucket-cloud-common module with the integration module though. I would rather keep it decoupled.

Reply:
Any special reason? It's a core library after all.

Reply:
I have replaced it by using the integrations module